src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,135 @@
+package duke.ui;
+
+import duke.data.messages.Messages;
+
+/**
+ * Class that deals with interactions with the user of Duke.
+ *
+ * @author Won Ye Ji
+ */
+public class Ui {
+
+    /**
+     * Greets the user.
+     */
+    public static void greet() {
+        System.out.println(Messages.WELCOME_MESSAGE);
+    }
+
+    /**
+     * Farewell message to the user.
+     */
+    public static void bye() {
+        System.out.println(Messages.INDENTATION + Messages.BYE_MESSAGE);
+    }
+
+    /**
+     * Informs the user that they have inputted an empty task.
+     * 
+     * @param s Type of task input.
+     * @return The error message to the user.
+     */
+    public static String emptyDescription(String s) {
+        String error = "OOPS!!!";
+        switch(s) {
+        case "todo":
+            error = Messages.INDENTATION + Messages.EMPTY_TODO_DESC;
+            break;
+        case "deadline":
+            error = Messages.INDENTATION + Messages.EMPTY_DEADLINE_DESC;
+            break;
+        case "event":
+            error = Messages.INDENTATION + Messages.EMPTY_EVENT_DESC;
+            break;
+        case "find":
+            error = Messages.INDENTATION + Messages.EMPTY_FIND_CMD;
+            break;
+        }
+        return error;
+    }
+
+    /**
+     * Informs the user that they have entered an unidentifiable command.
+     *
+     * @return Error message to the user.
+     */
+    public static String inputUnknown() {
+       return Messages.INDENTATION + Messages.UNKNOWN_INPUT_MESSAGE;
+    }
+
+    /**
+     * Informs the user that they have not entered a valid date.
+     *
+     * @return Error message to the user.
+     */
+    public static String dateMissing() {
+        return Messages.INDENTATION + Messages.MISSING_DATE;
```

Comment:
I think that you could add another method to add indentation to a String? Since you have `Messages.INDENTATION + ...` repeated many times throughout this class.